### PR TITLE
Allow custom resources to access all other resources

### PR DESCRIPTION
**Pull request by jquick**
Currently custom resources can only access core resources through the use of `inspec.*`. This change loads in any other custom resources so you can access them the same way.

Fixes https://github.com/inspec/inspec/issues/3107

Signed-off-by: Jared Quick <jquick@chef.io>

### CODE REVIEW

File: examples/custom-resource/controls/example.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# encoding: utf-8
+
+describe gordon do
+  its('example') { should_not eq 'foobar' }
```
Comment (clintoncwolfe) [inline examples/custom-resource/controls/example.rb]: My main concern is to use meaningful names.  Gordon here is meant to be Commissioner Gordon, from Batman.  What properties might he be dealing with?

its('crime_rate') { should be < 2 }
it { should have_a_fabulous_mustache }

I don't know what the 'example' property is, or why its value should or should not be  'foobar'.

File: examples/custom-resource/libraries/foobar.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+class Foobar < Inspec.resource(1)
```
Comment (clintoncwolfe) [inline examples/custom-resource/libraries/foobar.rb]: Perhaps Batsignal?

File: examples/custom-resource/libraries/foobar.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  name 'foobar'
+
+  example "
+      describe foobar do
+        its('info') { should eq 'foobar' }
+      end
+  "
+
+  def info
```
Comment (clintoncwolfe) [inline examples/custom-resource/libraries/foobar.rb]: Here, I'd suggest something like `whoami_output` or something.  Make it clearer that we're doing a neat trick - calling a core resource from a custom one.

File: examples/custom-resource/libraries/gordon.rb [hunk 4]
```diff
@@ -0,0 +1,15 @@
+class Gordon < Inspec.resource(1)
+  name 'gordon'
+
+  example "
+    describe gordon do
+      its('example') { should eq 'foobar' }
+    end
+  "
+
+  def example
```
Comment (clintoncwolfe) [inline examples/custom-resource/libraries/gordon.rb]: Likewise, maybe give this a name that makes it clear we're calling a custom resource from another custom resource.

Ooh, you could have a Batman and Bruce Wayne resource, and bruce implements `full_name` (which calls whoami) and Batman implements `secret_identity` (which calls full name on Batman)

Comment (clintoncwolfe) [inline examples/custom-resource/libraries/gordon.rb]: Also: defining a method named `example` is a bit high risk - `example` is part of the resource DSL!  No collision in this case, but the unfamiliar user (for whom this code is intended) would be easily confused.

**Review by clintoncwolfe — DISMISSED** [review]
The implementation code is great, no complaints.

Just some naming issues in the example code.

**Comment by jquick** [comment]
Thanks @clintoncwolfe - Updated the example to be a bit more understandable.

**Review by miah — APPROVED** [review]
APPROVE

**Review by clintoncwolfe — APPROVED** [review]
Looks good, thanks @jquick

**Pull request merged**